avm/ptn/deployment-script/import-image-to-acr/main.bicep

```
@@ -0,0 +1,246 @@
+param subnetResourceIds string[]?
+
+@description('Optional. Tags of the resource.')
+@metadata({
+  example: '''
+  {
+      "key1": "value1"
+      "key2": "value2"
+  }
+  '''
+})
+param tags object?
+
+// ============== //
+// Variables      //
+// ============== //
+
+var useExistingManagedIdentity = length(managedIdentities.?userAssignedResourcesIds ?? []) > 0
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.deploymentscript-importimagetoacr.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource acr 'Microsoft.ContainerRegistry/registries@2023-07-01' existing = {
+  name: acrName
+}
+
+// needed to "convert" resourceIds to principalId
+resource existingManagedIdentities 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = [
+  for resourceId in (managedIdentities.?userAssignedResourcesIds ?? []): if (assignRbacRole) {
+    name: last(split(resourceId, '/'))
+    scope: resourceGroup(split(resourceId, '/')[2], split(resourceId, '/')[4]) // get the resource group from the managed identity, as it could be in another resource group
+  }
+]
+
+resource newManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = if (!useExistingManagedIdentity && assignRbacRole) {
+  name: managedIdentityName ?? 'id-ContainerRegistryImport'
+  location: location
+  tags: tags
+}
+
+// assign the Contributor role to the managed identity (new or existing) to import images into the ACR
+resource acrRoleAssignmentExistingManagedIdentities 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for i in range(0, length(assignRbacRole ? (managedIdentities.?userAssignedResourcesIds ?? []) : [])): if (useExistingManagedIdentity) {
+    name: guid('roleAssignment-acr-${existingManagedIdentities[i].name}')
+    scope: acr
+    properties: {
+      principalId: existingManagedIdentities[i].properties.principalId
+      roleDefinitionId: subscriptionResourceId(
+        'Microsoft.Authorization/roleDefinitions',
+        'b24988ac-6180-42a0-ab88-20f7382dd24c'
+      )
+      principalType: 'ServicePrincipal' // See https://docs.microsoft.com/azure/role-based-access-control/role-assignments-template#new-service-principal to understand why this property is included.
+    }
+  }
+]
+resource acrRoleAssignmentNewManagedIdentity 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!useExistingManagedIdentity) {
```

```suggestion
resource acrRoleAssignmentNewManagedIdentity 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (assignRbacRole && !useExistingManagedIdentity) {
```

Should also be checked, right?